server/services/v1/auth.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/fullstorydev/grpchan/inprocgrpc"
+	"github.com/go-chi/chi/v5"
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/rs/zerolog/log"
+	api "github.com/tigrisdata/tigris/api/server/v1"
+	"github.com/tigrisdata/tigris/server/config"
+	"google.golang.org/grpc"
+)
+
+const (
+	oauthPattern = "/oauth/*"
+	refreshToken = "refresh_token"
+	scope        = "offline_access openid"
+)
+
+type authService struct {
+	api.UnimplementedAuthServer
+	OAuthProvider
+}
+
+type OAuthProvider interface {
+	GetAccessToken(request *api.GetAccessTokenRequest) (*api.GetAccessTokenResponse, error)
+}
+
+type Auth0Oauth struct {
```

Comment:
The name looks funny :)

Reply:
May be just Auth0 would work.

Reply:
renamed. I agree auth0 name is much simpler here.